jmh-tests/src/main/java/org/eclipse/collections/impl/jmh/AggregateByTest.java

```diff
@@ -678,12 +678,12 @@ public boolean equals(Object o)
         public int hashCode()
         {
             int result = (int) (this.getCount() ^ (this.getCount() >>> 32));
-            long temp = Double.doubleToLongBits(this.getSum());
-            result = 31 * result + (int) (temp ^ (temp >>> 32));
-            temp = Double.doubleToLongBits(this.getMin());
-            result = 31 * result + (int) (temp ^ (temp >>> 32));
-            temp = Double.doubleToLongBits(this.getMax());
-            result = 31 * result + (int) (temp ^ (temp >>> 32));
+            long temp1 = Double.doubleToLongBits(this.getSum());
+            result = 31 * result + (int) (temp1 ^ (temp1 >>> 32));
+            long temp2 = Double.doubleToLongBits(this.getMin());
+            result = 31 * result + (int) (temp2 ^ (temp2 >>> 32));
+            long temp3 = Double.doubleToLongBits(this.getMax());
+            result = 31 * result + (int) (temp3 ^ (temp3 >>> 32));
```

Comment:
I like all the fixes, not sure about the suppressions.

Reply:
Fixed all suppressions, but it does impact control flow, please review.

Reply:
Not sure what you mean by affecting control flow. The ones with reverse iteration?

Reply:
Yes the one with reverse iteration:
Basically we cap the iteration at 0.
https://github.com/eclipse/eclipse-collections/pull/611/commits/86ea7d1e0889036ce22c49297aa233bc875a7d9f#diff-1a2b9df9d2e5582ddda1ef2be9dba77bR332

https://github.com/eclipse/eclipse-collections/pull/611/commits/86ea7d1e0889036ce22c49297aa233bc875a7d9f#diff-f73899c981d66f3bf32157aca70650ecR391

Also, the compensation for Kahan summation algorithm: 
https://github.com/eclipse/eclipse-collections/pull/611/commits/86ea7d1e0889036ce22c49297aa233bc875a7d9f#diff-9d2319fbefc4b0d92c83470e981523bfR1397